### DISCUSSION TITLE
[Feature Request]: MariaDB

### DISCUSSION THREAD

**Discussion by Tklaversma** [beyondcode/herd-community#489] [Ideas]
Would be nice if we could chose MariaDB instead of MySQL.

Also being able to switch between versions (for all services) would be nice.

For example:
- MariaDB 10.4, 10.5, etc...

**Comment by Tklaversma** [comment]
@sschlein you got any update regarding this feature request?

**Comment by draske79** [comment]
Really looking forward to having this. All our projects are MariaDB

**Comment by sschlein** [comment]
MariaDB is coming soon!

**Reply by draske79** [reply]
Super! Great news

**Reply by Tklaversma** [reply]
@sschlein With "very soon", you didn't mean 1.7.0? According to the changelog its not there. Will it be the next update?

**Reply by sschlein** [reply]
We rescheduled some things to refactor some parts of the app to be able to deliver new features faster and so the services update is scheduled for 1.9 and targeted to early July.

1.8 will be the new UI for managing sites (Windows preview: https://twitter.com/DianaWebdev/status/1795827551849631931, adapted to macOS design, etc)

1.9 is Services with MariaDB and MongoDB and probabaly some more

You can still use your existing MariaDB instances with Herd, I assume that you run them via brew?

**Reply by Tklaversma** [reply]
@sschlein Thanks for the update. Much appreciated.

No, I'm still using MAMP Pro. Herd up to this point is basically a slim rebuild of MAMP (just install and it works, has all the features Herd has and much more).

But MAMP is not being actively maintained anymore, as far as I can tell. So switching to the next best thing would be Herd, whenever we get MariaDB and Apache support.

**Reply by sschlein** [reply]
Oh, Apache is not supported and probabaly never will be 🙈

**Reply by Tklaversma** [reply]
@sschlein I figured. That's to bad though, since many developers are still maintaining older projects of their customers (including me). So this means you exclude a very large group, but I understand the choice from your perspective.

Once Herd is at the level of MAMP (excluding Apache), I'll switch all projects to nginx (always a win) and I get to move to Herd, which is actively maintained. Win-win...

**Reply by tomcoonen** [reply]
I hope we also get version selection support for the services 💪

**Reply by sschlein** [reply]
I depends, if there are major differences, we'll definitely add this like we did with PostgreSQL but if there are just minor differences, we may skip versions

**Reply by tomcoonen** [reply]
Of course, I'm also not interested in patches. Minors and majors would be nice though since we like to work in sync with production environments (using Docker for that now).

**Reply by philbraun** [reply]
> Once Herd is at the level of MAMP (excluding Apache), I'll switch all projects to nginx (always a win) and I get to move to Herd, which is actively maintained. Win-win...

MAMP (Pro) is still maintained, they recently published 7.0.0 followed by 4 point updates.
Unfortunately, their Cloud backup solution has never worked for me with larger sites and using OneDrive (uploads silently failing), and today I tried migrating 4 client DEV sites from Herd to MAMP Pro, resulting in MAMP shooting itself and being frozen on start, with no recourse that I could find.

Since Herd also is faster overall and certainly a lot more stable, I've given up on MAMP (again), even though I find their pricing for existing customers at around €50/year more reasonable than Herd Pro's 100/y, which I'd only need for MariaDB (somehow still missing in 1.9.1?).

**Reply by Tklaversma** [reply]
@philbraun I've updated to MAMP Pro 7 last week. Works perfectly. I just need a well all-rounded solution so I can address all my customers with one tool, which Herd can't (yet?).

**Reply by philbraun** [reply]
@Tklaversma Happy it works for you! I upgraded and then tried MAMP Pro 7 for 10 days now, and would've kept it despite the Cloud backup bugs, but today's unrecoverable crash really soured me on it.

**Comment by grzegorzkibitz** [comment]
Any update on this? 😆

**Comment by sschlein** [comment]
Still planned 🙈 As you might got from the recent Laracon US reveal, there are huge improvements coming to Herd and building these required all our focus. After this release, we'll finalize the update for the services that include MariaDB.

**Reply by draske79** [reply]
As I am not in the US, can I find somewhere these Laracon US reveals with the announced huge improvements?

**Reply by sschlein** [reply]
We'll release them today for macOS and soon for Windows. After that, that's on the list but it's not our highest priority right now.

**Reply by draske79** [reply]
Do I understand MariaDB is not the highest priority right now? As it is the second most voted feature. And it is a paid feature.

**Reply by sschlein** [reply]
To be honest, 12 votes of tens of thousands of daily users is not a majority 🙈

If MariaDB is important for you, you can install it via dbngin or by downloading it directly and it just works. We'll eventually make it available via Herd Pro as a one click installer but that will take a few weeks from now.

**Reply by draske79** [reply]
I'm a little disappointed with the tone of voice in that reply. 

I personally would love to see MariaDB in Herd. But, "no", or "end of the year" is also an answer. I was just wondering what the big improvements were in Herd that were announced in the US, which apparently got priority over this. And after reading your sentence, "After that, that's on the list, but it's not our highest priority right now." I was wondering what "that's" referred to.

Anyhow, your answer shows that voting for features or asking feedback has no effect or purpose, so I will stop doing that. Have a nice day and good luck with your product!

**Reply by sschlein** [reply]
I'm sorry but I promise that you'll get more of the update than of MariaDB as a one click installer. We've worked for months on this update and the new features that drop later today are:

* Forge Integration: Connect local sites with Herd and deploy them from Herd
* Profiler: Identify bottlenecks in your application
* Herd.yml: Share site configurations with your team
* Herd Services: Launch secure Reverb instances with effortless TLS certificates
* Dump Debugger: Dump Eloquent queries, HTTP requests, logs, jobs and Blade views without installing a package


We've also rewritten the whole documentation and fixed countless minor issues.

**Reply by kevinpickett** [reply]
Just a quick clarification, MariaDB is not available via DBngin and tends to be a huge pain point. 

I can't speak for all, but my team has been anxiously watching this discussion. Knowing it was one of the most upvoted of all time, we did not think we needed to apply further pressure to make this important. We figured the pressure was already there, the community had spoken and we had been told we would receive it. 

I think it is a bit disingenuous to presume that your tens of thousands of users are going to find these discussions for a closed source product. Of course we aren't asking for it day after day, we were told 98 days ago it would be here soon. If the community votes on here are useless, is there another way we should be escalating our needs as a community?

I definitely don't want this to be some type of ultimatum (my team is small and probably wont hardly register on overall sales) but our VPE will not approve our Herd Team license purchase request until MariaDB is included. With today's release being the 3rd release since we were told soon, the tone of the conversation on my team shifted from "wait for it" to "time to look elsewhere". 

Outside of my employment, I paid for Herd Pro personally. It is an excellent product that has drastically improved DX on my personal projects. I can't help but pass this information along because your update here today comes through as tone deaf and will needlessly push way enterprise customers paying attention to a young and exciting product.

**Reply by Tklaversma** [reply]
@kevinpickett I gave up shifting to Herd and updated to MAMP Pro 7.  It has all you need. I wanted to go along with the Laravel community and support beautiful products like this, but like you said, what's the point when the community is not being Herd (pun intended). Glad to know I'm not the only one with these thought. Thanks for sharing.

**Reply by philbraun** [reply]
@Tklaversma Unfortunately, I don't think MAMP Pro 7 has MariaDB, which is what this thread is about. (Or did I miss it?).
Also, currently MAMP Pro 7 has a pretty severe bug that limits how many DB tables you can have in all databases to 2524 – if you have 2525 or more, MAMP will immediately freeze.

**Reply by Tklaversma** [reply]
@philbraun never ran into that problem of too many tables overall... good to know. 

About MariaDB, I'm running it via MAMP. Just brew MariaDB and make a symbolic link if I remember correctly (it has been over 3 years since I set it up).

**Reply by philbraun** [reply]
Came back here to check for news re: MariaDB in Herd (Pro), since MAMP 7 Pro a) also does not have MariaDB and b) unfortunately still is really slow, at least for Wordpress (anything in MAMP basically takes twice as long as on a run-of-the-mill shared server)... but apparently «a few weeks from now» takes about as long for @kevinpickett as it usually does for me :-)

**Comment by RoodFruit** [comment]
> After this release, we'll finalize the update for the services that include MariaDB.

@sschlein any update on this after your post on August 29th?

**Comment by maxwoerner** [comment]
@sschlein Is there any update on this? Would love to use MariaDB as a Herd service...

**Reply by igorpreston** [reply]
@sschlein support for MariaDB will definitely get me as a paid user - I use MariaDB with Laravel extensively and have to manage installations separately for now. Really looking forward for this feature landing on the release!

**Comment by veltix** [comment]
@sschlein  Any updates?

**Comment by bsterckx** [comment]
Seems to be coming in the next herd release: 
![image](https://github.com/user-attachments/assets/756b777c-40d3-42c5-97dd-86cda26b7f91)

[Check post on x](https://x.com/marcelpociot/status/1894158825164268015)

**Reply by Tklaversma** [reply]
Glad its finally being added.. but they missed the boat for me. Happily running MAMP Pro 7.

**Comment by bsterckx** [comment]
It's here for both Windows & MacOS v1.18.0: [changelog](https://herd.laravel.com/changelog/macos).
It seems to installs the 10.11 version.
